### PR TITLE
br-profile: created profile ui

### PR DESCRIPTION
**Pull request by melievkh**

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  Text,
+  StyleSheet,
+  ActivityIndicator,
+  TouchableOpacityProps,
+  ViewStyle,
+  TextStyle,
+} from 'react-native';
+import { useTheme } from 'src/theme';
+
+interface CustomButtonProps extends TouchableOpacityProps {
```
Comment (hgoziev) [inline src/components/Button/Button.tsx]: Let's create a separate file for types. For example: `Button.types.ts`

File: src/components/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  title,
+  onPress,
+  isLoading,
+  variant,
+  disabled,
+  style,
+  ...restProps
+}) => {
+  const { colors } = useTheme();
+  const getButtonStyle = (): ViewStyle => {
```
Comment (hgoziev) [inline src/components/Button/Button.tsx]: Let's move this helper functions to a separate file. If these are relevant to Button.tsx only, then we should create `Button.utils.tsx` file and place these functions there.

File: src/components/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        <ActivityIndicator size="small" color="white" />
+      ) : (
+        <Text style={getButtonTextStyle()}>{title}</Text>
+      )}
+    </TouchableOpacity>
+  );
+};
+
+const baseButtonStyles: ViewStyle = {
+  paddingVertical: 10,
```
Comment (hgoziev) [inline src/components/Button/Button.tsx]: We can create a predefined padding values like xxs, xs, s, m, l, xl, xxl => 5, 10, 15

File: src/components/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  paddingHorizontal: 15,
+  borderRadius: 5,
+  alignItems: 'center',
+};
+
+const styles = StyleSheet.create({
```
Comment (hgoziev) [inline src/components/Button/Button.tsx]: Let's move these styles into a separate file. For example: `Button.styles.js`.

File: src/components/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  button: {
+    ...baseButtonStyles,
+  },
+  buttonText: {
+    color: '#ddd',
```
Comment (hgoziev) [inline src/components/Button/Button.tsx]: We should have a predefined color template values and we should use them

File: src/components/Button/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    fontSize: 18,
```
Comment (hgoziev) [inline src/components/Button/Button.tsx]: The same with FontSize as well. We should have a set of font size values defined in one place and used in different places with their names.

File: src/components/Inputs/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+interface InputProps extends TextInputProps {
+  label?: string;
+  icon?: 'person-outline' | 'local-phone';
+  error?: string;
+}
+
+const Input: React.FC<InputProps> = ({ label, icon, error, ...rest }) => {
+  return (
+    <View style={styles.container}>
+      {label && <Text style={styles.label}>{label}</Text>}
```
Comment (hgoziev) [inline src/components/Inputs/Input.tsx]: Do you have designs ready ? If you have it, can you share it to me ?

File: src/components/Inputs/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+      <View style={styles.inputContainer}>
+        {icon && <MaterialIcons name={icon} size={24} style={styles.icon} />}
```
Comment (hgoziev) [inline src/components/Inputs/Input.tsx]: Icons could be on both sides. It's better to name it leftIcon and rightIcon.

File: src/components/ProfileHeader/ProfileHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,133 @@
+import { Ionicons } from '@expo/vector-icons';
+import { NavigationProp, useNavigation } from '@react-navigation/native';
+import { StyleSheet, TouchableOpacity, Image } from 'react-native';
+
+import { View } from '../ThemedView/ThemedView';
+import Text from '../Text/Text';
+import Button from '../Button/Button';
+
+export default function ProfileHeader({ isLoggedIn }: { isLoggedIn: boolean }) {
+  const navigation = useNavigation<NavigationProp<any>>();
```
Comment (hgoziev) [inline src/components/ProfileHeader/ProfileHeader.tsx]: Let's not use `any` as type if possible.

File: src/components/ProfileHeader/ProfileHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+      ) : (
+        <View style={styles.loggedInHeader}>
+          <View style={styles.titleWrapper}>
+            <Text style={styles.title}>Profile</Text>
+            <Ionicons name="notifications-outline" size={26} />
+          </View>
+
+          <TouchableOpacity
+            style={styles.redirectToProfileWrapper}
+            onPress={() => navigation.navigate('redirectToCreate')}
```
Comment (hgoziev) [inline src/components/ProfileHeader/ProfileHeader.tsx]: All route names must be a constant value. For example: `routes.SIGNUP`

File: src/components/ProfileHeader/ProfileHeader.tsx [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+          >
+            <View style={styles.userInfosWrapper}>
+              <View style={styles.profileImageWrapper}>
+                <Image
+                  source={{
+                    uri: 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcRYR2tzlAo8Mr4r_1MNOOVAsW-NXHnGmCuQwGTfda_x9RBu6YWP6X1ohBGSVtIHo4rzKjw&usqp=CAU',
```
Comment (hgoziev) [inline src/components/ProfileHeader/ProfileHeader.tsx]: This uri value should be in a constant value even it is  a mock data.

File: src/navigation/stacks/ProfileStacks/CreateProfileStack.tsx [hunk 12]
```diff
@@ -0,0 +1,167 @@
...
+  Platform,
+  ScrollView,
+  KeyboardAvoidingView,
+} from 'react-native';
+import { Button, Input, Text, View } from 'src/components';
+import { ScreenTemplate } from 'src/templates';
+
+export default function CreateProfileStack() {
+  const genderOptions = ['Male', 'Female'];
+  const [fullname, setFullname] = useState('');
```
Comment (hgoziev) [inline src/navigation/stacks/ProfileStacks/CreateProfileStack.tsx]: It's better to have a single useState object value to store all these information

File: src/navigation/stacks/ProfileStacks/CreateProfileStack.tsx [hunk 13]
```diff
@@ -0,0 +1,167 @@
...
+            />
+            <TouchableOpacity style={{ ...styles.addPhotoButton }}>
+              <Ionicons name="add" size={24} />
+              <Text>add photo</Text>
+            </TouchableOpacity>
+          </View>
+        </View>
+        <KeyboardAvoidingView
+          behavior={Platform.OS === 'ios' ? 'padding' : 'height'}
+          style={{ gap: 10 }}
```
Comment (hgoziev) [inline src/navigation/stacks/ProfileStacks/CreateProfileStack.tsx]: Let's move to a separate styling file.

File: src/navigation/stacks/ProfileStacks/CreateProfileStack.tsx [hunk 14]
```diff
@@ -0,0 +1,167 @@
...
+        >
+          <View>
+            <Text style={{ fontSize: 24, fontWeight: 'bold' }}>Profile</Text>
```
Comment (hgoziev) [inline src/navigation/stacks/ProfileStacks/CreateProfileStack.tsx]: All Styles should be in a separate file (no matter how small ) and  used in actual component file.

File: src/screens/ProfileScreen/ProfileScreen.tsx [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+        ListHeaderComponent={<ProfileHeader isLoggedIn={isLoggedIn} />}
+        data={[
+          {
+            label: 'Settings',
+            data: isLoggedIn ? profileSectionData.whenLoggedIn : profileSectionData.whenLoggedOut,
+          },
+          { label: 'Help', data: profileSectionData.help },
+        ]}
+        keyExtractor={(item, index) => index.toString()}
+        renderItem={({ item }) => (
```
Comment (hgoziev) [inline src/screens/ProfileScreen/ProfileScreen.tsx]: `renderItem` and `ListFooterComponent` should be a separate function or component and should be used like `ListHeaderComponent`

File: src/theme/index.tsx [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+  const defaultTheme: ThemeContextProps = {
+    isDark,
+    colors: isDark ? darkColors : lightColors,
+    setScheme: (scheme: 'light' | 'dark') => setIsDark(scheme === 'dark'),
+  };
+
+  return <ThemeContext.Provider value={defaultTheme}>{children}</ThemeContext.Provider>;
+};
+
+export const useTheme = () => React.useContext(ThemeContext);
```
Comment (hgoziev) [inline src/theme/index.tsx]: `useTheme` custom hook should handle an error.  

`const context = useContext(ThemeContext)`
`if(!context) throw error ... `
`return context`

File: src/components/Inputs/Input.tsx (hunk 7 above)
Comment (melievkh) [reply src/components/Inputs/Input.tsx]: you mean figma design or the design that I've created?

File: src/theme/index.tsx
Comment (javokhirm) [inline src/theme/index.tsx:1]: no need to use context, include theme on redux-toolkit

File: src/styles/containers.ts [hunk 17]
```diff
@@ -0,0 +1,14 @@
+export enum SIZES {
+  SM = 'sm',
+  MD = 'md',
+  LG = 'lg',
+}
+
+export const FONT_SIZES = {
```
Comment (javokhirm) [inline src/styles/containers.ts]: seems weird, why do you need FONT_SIZES as constant? better create a separate file as typography.ts and include common text styles

File: src/types/index.ts
Comment (javokhirm) [inline src/types/index.ts:1]: create dedicated files for each type, and use index.ts files as exporter files as you have done on many folders

File: src/api/store/store.ts
Comment (javokhirm) [inline src/api/store/store.ts:1]: `store` folder should be located in `src` folder, not inside `api` folder

File: src/api/types.ts
Comment (javokhirm) [inline src/api/types.ts:1]: we have `types` folder to include all types

**Review by javokhirm — CHANGES_REQUESTED** [review]

**Review by javokhirm — APPROVED** [review]

**Pull request merged**